packages/headless/src/api/knowledge/stream-answer-api.ts

```diff
@@ -371,6 +372,32 @@ export const constructAnswerQueryParams = (
       firstResult: state.pagination.firstResult,
     }),
     tab: selectActiveTab(state.tabSet),
+    locale: state.configuration.search.locale,
+    timezone: state.configuration.search.timezone,
+    ...(state.debug !== undefined && {debug: state.debug}),
+    ...(navigatorContext.referrer?.length && {
+      referrer: navigatorContext.referrer,
+    }),
+    actionsHistory: state.configuration.analytics.enabled
+      ? HistoryStore.getInstance().getHistory()
+      : [],
+    ...(state.folding && {
+      filterField: state.folding.fields?.collection,
+      childField: state.folding.fields?.parent,
+      parentField: state.folding.fields?.child,
+      filterFieldRange: state.folding.filterFieldRange,
+    }),
+    ...(state?.excerptLength?.length && {
+      excerptLength: state.excerptLength.length,
+    }),
+    ...(state.dictionaryFieldContext &&
+      !!Object.keys(state.dictionaryFieldContext.contextValues).length && {
+        dictionaryFieldContext: state.dictionaryFieldContext.contextValues,
+      }),
```

Comment:
I think there is an opportunity here to leverage selectors for these. Its not only best practice, but if you search for these fields in the repo, its also set in other places.

Especially the fields where you have some logic before setting the fields are prime candidates to be used with selectors IMO. What do you think?